controller/gateway/controller_cleanup.go

```diff
@@ -140,13 +149,14 @@ func handleGatewayFinalizerPatchOrUpdateError(err error, gateway *gatewayv1.Gate
 		return ctrl.Result{}, nil
 	}
 
-	// If the Gateway is not found, then requeue without an error.
-	if k8serrors.IsNotFound(err) {
+	// If the Gateway is not found or there's a conflict patching, then requeue without an error.
+	if k8serrors.IsNotFound(err) || k8serrors.IsConflict(err) {
```

Comment:
I'm not sure we want to silently return in case of a conflict. It can be useful to understand what's going on from time to time. WDYT?

Reply:
What value do you think, the conflict error would give to users?

Since the controller is the one that manages these entities, at the end of the day, it is what causes the conflict to be there in the first place. By logging those errors we're letting the user know that "hey, the controller that you're using observed a conflict. You don't need to act on it, but it's an error."

Reply:
Rather than being useful for users, I find this kind of error useful when debugging. But I see your point, let's keep it as it is 👍